File: 2015/05/part_2.go
// Usage: From root folder
// $ go run 2015/05/part_2.go 2015/05/input

package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func rule1(s string) bool {
	for i := 0; i < len(s)-2; i++ {
		if strings.Count(s, s[i:i+2]) >= 2 {
			return true
		}
	}

	return false
}

func rule2(s string) bool {
	for index := range s {
		if index >= 1 && index < len(s)-1 {
			if s[index-1] == s[index+1] {
				return true
			}
		}
	}

	return false
}

func isNiceString(s string) bool {
	return rule1(s) && rule2(s)
}

func main() {
	file, err := os.Open(os.Args[1])
	if err != nil {
		os.Exit(1)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	niceStringsCount := 0
	for scanner.Scan() {
		if isNiceString(scanner.Text()) {
			niceStringsCount++
		}
	}

	fmt.Println(niceStringsCount)
}

File: 2021/01/part_2.py
#  Usage: From root folder
#  $ python 2021/01/part_2.py 2021/01/input

import sys


if __name__ == "__main__":
    with open(sys.argv[1]) as file:
        measurements = [int(m) for m in file.readlines()]

    before = sum(measurements[0:3])
    count = 0
    for i in range(len(measurements)-2):
        current = sum(measurements[i:i+3])
        if current > before:
            count += 1
        before = current

    print(count)
